### PR TITLE
1 add command line arg for directory path

### PR DESCRIPTION
**Pull request by furcon**
@drayside @a2ptk @eliavhyehuda @lynnegithub @ja4green 
Code review and then we're good to merge.
Cheers,
fjq

### CODE REVIEW

**Review by eliavhyehuda — APPROVED** [review]

**Review by ishpattt — APPROVED** [review]
LGTM, was able to get it working locally.

**Comment by lynnegithub** [comment]
🚢

**Review by lynnegithub — APPROVED** [review]

**Comment by drayside** [comment]
This does not work for me locally. Observations:

1. It does appear to accept the command line argument. I added a line to print that to the console. But it appears that it does not actually use this argument to load the markdown files.

2. The Overview/Index page is still clearly loading the files from config/markdown, even when I specify another directory. I tested this by modifying the tag "Afria" to "Africa" in config/markdown/blueprint-somo-project.md, and I can see this change reflected on the Overview page.

3. When I click on individual projects, I get a blank page. This blank page occurs whether I specify the markdown command line argument or not.

**Comment by furcon** [comment]
Please ignore that a large part of the diff for 'config/parse.js' file shows indentation changes. I parameterized the 'initParse' function so it takes the markdown directory path.

**Comment by drayside** [comment]
This now partially works for me:

1. The Overview/Index page appears to be working. It is reading the correct directory.

2. If I click on an individual project, then the page is blank. Doesn't matter which project.

**Comment by drayside** [comment]
Now throws an exception when I try to start up the server:

`$ nodejs server.js --markdown-directory $PWD/../2018/projects/
 --- Configuring for Local Dev --- 
--- Using specified markdown directory path: '/home/drayside/code/svn.ece/se_capstone/2018/projects' ---
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE :::3000
    at Object.exports._errnoException (util.js:870:11)
    at exports._exceptionWithHostPort (util.js:893:20)
    at Server._listen2 (net.js:1237:14)
    at listen (net.js:1273:10)
    at Server.listen (net.js:1369:5)
    at Server.listen (/home/drayside/code/svn.ece/se_capstone/se-capstone-db/node_modules/restify/lib/server.js:314:32)
    at Object.<anonymous> (/home/drayside/code/svn.ece/se_capstone/se-capstone-db/server.js:86:8)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Function.Module.runMain (module.js:442:10)
    at startup (node.js:136:18)
    at node.js:966:3
`

File: app/routes/projectHandlers.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ module.exports = function (projectHelpers, markdownDirectory) {
 
     var compileMarkdown = function compileMarkdown(req, res, next){
         var projectName = req.params.projectName;
-        var fpath = path.join(markdownDirectory, projectName + '.md'); //fname = ${projectName}.md TODO - confirm this is always true
+        var fpath = path.join(markdownDirectory, projectName);
```
Comment (furcon) [inline app/routes/projectHandlers.js:30]: This change caused this route https://github.com/drayside/se-capstone-db/blob/1-add-command-line-arg-for-directory-path/server.js#L79 to fail, now when I remove out the `+ .md ` the route works. But another route https://github.com/drayside/se-capstone-db/blob/1-add-command-line-arg-for-directory-path/server.js#L63 fails. I can't understand how the first route (first link) can cause compileMarkdown to be excecute. I'm perplexed.

**Review by furcon — COMMENTED** [review]
Perplexed.

**Comment by drayside** [comment]
Sorry, that stack trace in my last post (now deleted) was due to already having the server running in another window.

**Comment by furcon** [comment]
@drayside Is the `/project/html/[project-name]` route needed ?

**Comment by furcon** [comment]
I get it now, so `/project/:[prrojectname]` routes to serverIndex which calls `next()` which matches to the next matching route which is `/project/html/:projectname`

**Pull request merged**